eng/scripts/automation/Invoke-DataPlaneGenerateSDKPackage.ps1

```diff
@@ -22,17 +22,10 @@ $outputJson = Get-Content $outputJsonFile | Out-String | ConvertFrom-Json
 $projectFolder = $outputJson.projectFolder
 Write-Host "projectFolder:$projectFolder"
 Remove-Item $outputJsonFile
+# Generate Code
 Invoke-Generate -sdkfolder $projectFolder
-if ( $? -ne $True) {
-  Write-Error "Failed to create generate sdk."
-  exit 1
-}
-
```

Comment:
So no error checking here? Alternatively, I often start my PowerShell scripts with `$ErrorActionPreference = 'Stop'` so any PowerShell error within the scripts or other scripts/commands it calls halts script execution.

Reply:
1. I explained in a previous comment that the lowercase package directory and project file names will impact the output name. We don't want to publish packages like Azure.test.sample. It looks bad, and is inconsistent with our package formatting. Your template config needs separate replacements for case-sensitive replacements, as well as case-sensitive replacements with lowercase replacement text just like I have in my original template config.
2. So then how do you copy the ci.yml, Directory.Build.props, test-resources.json, and tests.yml files in the service directory? I tried creating a template in different directories and never saw this important files.

Reply:
#1 Yes, we need to avoid lowercase package directory. But current, DPG eng script will use the package folder name as the namespace, even we force to use the tempelate.json to re-generate the package name, in the Generated Code, the namespace is also the package name, it will cause build fail. So we write script and enforce this constrains in it and describe it in the start doc.
#2 you can set --includeCI in the `dotnet new` command ( `dotnet new azsdkdpg -g Test -li Sample -s local.json --includeCI`), then those ci.yml, Directory.Build.props, test-resources.json, and tests.yml files will be copied into the service directory.

Reply:
Tip: don't use `#<num>` unless referring to an issue or pull request. It adds a link to these comments on those (very old) issues. Just use numbered lists e.g., `1.`, or quote and reply from the comment toolbar.

1. That sounds like a problem. Instead, I recommend the docs say to pass the service directory name and package name cased as they want it, and the template defines a separate variable with the lowercase name. That's what my previous template had and works just fine. The DPG script shouldn't be a factor. Instead of deriving the package name from the directory name, derive the directory name from the package name. Much easier to control casing. See my old template for an example. I recommend, like with your extension, you check out an older branch and run my extension to see how it works.
2. But those files should be there by default unless excluded or already there. Every track 2 library should have those files. Directory.Build.props is optional, but doesn't hurt and helps prevent teams from inventing their own ways when needed.